libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -424,13 +426,21 @@ void AC_PosControl::set_max_speed_accel_xy(float speed_cms, float accel_cmss)
     _vel_max_xy_cms = speed_cms;
     _accel_max_xy_cmss = accel_cmss;
 
-    // ensure the horizontal time constant is not less than the vehicle is capable of
-    const float lean_angle = _accel_max_xy_cmss / (GRAVITY_MSS * 100.0 * M_PI / 18000.0);
-    const float angle_accel = MIN(_attitude_control.get_accel_pitch_max(), _attitude_control.get_accel_roll_max());
-    if (is_positive(angle_accel)) {
-        _tc_xy_s = MAX(_shaping_tc_xy_s, 2.0 * sqrtf(lean_angle / angle_accel));
-    } else {
-        _tc_xy_s = _shaping_tc_xy_s;
+    // ensure the horizontal jerk is less than the vehicle is capable of
+    const float jerk_max_cmsss = MIN(_attitude_control.get_ang_vel_roll_max_rads(), _attitude_control.get_ang_vel_pitch_max_rads()) * GRAVITY_MSS * 100.0;
+    const float snap_max_cmssss = MIN(_attitude_control.get_accel_pitch_max_radss(), _attitude_control.get_accel_roll_max_radss()) * GRAVITY_MSS * 100.0;
```

Comment:
```suggestion
    const float snap_max_cmssss = MIN(_attitude_control.get_accel_roll_max_radss(), _attitude_control.get_accel_pitch_max_radss()) * GRAVITY_MSS * 100.0;
```

To match the line above

Reply:
Nice, thanks!